File: house/controllers/upload_controller.py
from house.controllers.basic_controller import *

class UploadController(BasicController):
    """docstring for LoadController."""
    def __init__(self, args):
        super().__init__(args)

    def draw(self):
        dirs = self.args[0].split('/')
        levels = 1 + len(dirs)
        commands = []

        commands.append('cd src')
        for direc in dirs:
            commands.append('cd ' + direc)
        commands.append('git add -A')
        commands.append('git commit')
        commands.append('git push origin master')
        for _ in range(levels):
            commands.append('cd ..')

        self.execute(commands)

File: house/controllers/upload_controller.lua
local basic_controller = require "house/controllers/basic_controller"
local upload_controller = { }

upload_controller.new = function(args)
  local self = basic_controller.new(args)

  self.draw = function()
    local commands = { }
    local repo = self.args[1]

    if repo ~= nil then
      commands = self.buildTree(repo, self.addCommands)
    else
      commands = self.addCommands(commands)
      table.insert(commands, #commands-2, "git checkout " .. self.script_name)
    end

    self.execute(commands)
  end

  self.addCommands = function(commands)
    table.insert(commands, "git add -A")
    table.insert(commands, "git commit")
    table.insert(commands, "git push origin master")
    return commands
  end

  return self
end

return upload_controller
